packages/eas-cli/src/update/utils.ts

```diff
@@ -74,3 +78,60 @@ export function ensureValidVersions(exp: ExpoConfig, platform: RequestedPlatform
     throw error;
   }
 }
+
+export async function checkDeprecatedChannelConfigurationAsync(
+  projectDir: string
+): Promise<boolean> {
+  const easJson = await new EasJsonReader(projectDir).readAsync();
+  if (easJson.build && Object.entries(easJson.build).some(([, value]) => value.releaseChannel)) {
+    Log.warn(`» One or more build profiles in your eas.json specify the "releaseChannel" property.
+For EAS Update, you need to specify the "channel" property, or your build will not be able to receive any updates.
+Update your eas.json manually, or run ${chalk.bold('eas update:configure')}.
+${learnMore('https://docs.expo.dev/eas-update/getting-started/#configure-your-project')}`);
+    Log.newLine();
+    return true;
+  }
+
+  return false;
+}
+
+export async function checkBuildProfileConfigMatchesProjectConfigAsync(
```

Comment:
The same as above, it should either handle the situation or return boolean(not both)